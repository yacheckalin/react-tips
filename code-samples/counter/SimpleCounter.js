import React, { useState } from "react";
import PropTypes from "prop-types";

const SimpleCounter = ({ initial }) => {
  const [counter, setCounter] = useState(initial);

  return (
    <div className="container">
      <div className="row">
        <div className="col s1">
          <span
            id="desc_counter"
            className="flow-text"
            onClick={(e) => setCounter(counter - 1)}
          >
            <i className="material-icons red-text">remove_circle_outline</i>
          </span>
        </div>
        <div className="col s2 blue-text" id="counter_value">
          {counter}
        </div>
        <div className="col s1">
          <span
            id="asc_counter"
            className="flow-text"
            onClick={(e) => setCounter(counter + 1)}
          >
            <i className="material-icons green-text">add_circle</i>
          </span>
        </div>
      </div>
    </div>
  );
};

SimpleCounter.propTypes = {
  initial: PropTypes.number.isRequired,
};

export default SimpleCounter;
