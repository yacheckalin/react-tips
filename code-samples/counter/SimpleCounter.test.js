import React from "react";
import toJSON from "enzyme-to-json";
import { shallow } from "enzyme";
import SimpleCounter from "./SimpleCounter";

describe("<SimpleCounter />", () => {
  test("displays and renders properly", () => {
    const wrapper = shallow(<SimpleCounter initial={0} />);

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  test("initial value shows properly", () => {
    const wrapper = shallow(<SimpleCounter initial={0} />);

    expect(wrapper.find("#counter_value").text()).toBe("0");
    wrapper.unmount();
  });

  test("ascending properly", () => {
    const wrapper = shallow(<SimpleCounter initial={0} />);

    wrapper.find("#asc_counter").simulate("click");
    expect(wrapper.find("#counter_value").text()).toBe("1");
    wrapper.unmount();
  });

  test("descending properly", () => {
    const wrapper = shallow(<SimpleCounter initial={10} />);

    wrapper.find("#desc_counter").simulate("click");
    expect(wrapper.find("#counter_value").text()).toBe("9");
    wrapper.unmount();
  });
});
