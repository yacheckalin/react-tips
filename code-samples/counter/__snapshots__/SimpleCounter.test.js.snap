// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SimpleCounter /> displays and renders properly 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col s1"
    >
      <span
        className="flow-text"
        id="desc_counter"
        onClick={[Function]}
      >
        <i
          className="material-icons red-text"
        >
          remove_circle_outline
        </i>
      </span>
    </div>
    <div
      className="col s2 blue-text"
      id="counter_value"
    >
      0
    </div>
    <div
      className="col s1"
    >
      <span
        className="flow-text"
        id="asc_counter"
        onClick={[Function]}
      >
        <i
          className="material-icons green-text"
        >
          add_circle
        </i>
      </span>
    </div>
  </div>
</div>
`;
